addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -30,6 +31,22 @@ class DiscussChannelMember(models.Model):
     is_self = fields.Boolean(compute="_compute_is_self", search="_search_is_self")
     # channel
     channel_id = fields.Many2one("discuss.channel", "Channel", ondelete="cascade", required=True, auto_join=True)
+    channel_ownership = fields.Selection(
+        _channel_ownership_list,
+        default=None,
+        string="Channel Role",
+        groups="base.group_system",
+    )
+    channel_ownership_in_view = fields.Selection(
+        _channel_ownership_list,
+        compute="_compute_channel_ownership_in_view",
+        inverse="_inverse_channel_ownership_in_view",
+        compute_sudo=True,
```

Comment:
If it works I prefer the sudo inside the compute on the specific part that needs it rather than sudo the whole compute.